mapusaurus/geo/views.py

```diff
@@ -23,12 +24,30 @@ def tract_centroids_in_json_format(request):
     return response % ', '.join(geo.tract_centroids_as_geojson() for geo in censusgeos)
 
 def get_censustract_geoids(request):
-    geos = get_censustract_geos(request)
+    if 'neLat' in request.GET:
+        geos = get_censustract_geos(request)
+    else:
+        geos = get_tracts_by_msa(request)
     return geos.values_list('geoid', flat=True)
 
-def get_censustract_geos(request):
+def get_tracts_by_msa(request):
+    msa_id = request.GET.get('metro')
+    metro = Geo.objects.filter(geo_type=Geo.METRO_TYPE, geoid=msa_id).first()
+    # lender_id = request.GET.get('lender')
+    # lender = Institution.objects.filter(institution_id=lender_id).first()
+    # if metro and lender:
+    if metro:
+        tracts = Geo.objects.filter(geo_type=Geo.TRACT_TYPE, cbsa=msa_id)
+        return tracts
+    else:
+        return []
+
```

Comment:
This doesn't follow the previous convention but can you move this function into the Geo class in models.py? Since its a "getter" method and not a view, it makes sense to put it into the class.